src/main/java/net/mcreator/ui/component/JColor.java

```diff
@@ -103,6 +96,21 @@ public JColor(Window window, boolean allowNullColor, boolean allowTransparency)
 		}
 	}
 
+	public static void openDialog(Window window, String dialogTitle, Color initColor, Consumer<Color> onColorSelected) {

```

Comment:
A more Javanic solution would be that openDialog returns Color instead of Consumer. dialog.setVisible(true); should block, so you can store selected color in AtomicObject<Color> and then return it.

This is more inline with how JOptionPane dialogs operate